docs/OSUpgrades.md

```diff
@@ -0,0 +1,74 @@
+# OS updates
+
+The MCO manages the Red Hat Enterprise Linux CoreOS operating system. Further,
+the operating system itself is just another part of the [release image](https://github.com/openshift/cluster-version-operator/), called `machine-os-content`.
+
+In other words, the cluster controls the operating system.
+
+# "Bootimage" vs machine-os-content
+
+We will use the term "bootimage" to mean an initial RHCOS disk image, such
+as an AMI, bare metal raw disk image, VMWare VMDK, OpenStack qcow2, etc.  
+These bootimages are built using [OSTree](https://github.com/ostreedev/ostree/)
+and [rpm-ostree](https://github.com/projectatomic/rpm-ostree/).
+
+Today, [the installer](https://github.com/openshift/installer/) pins the "bootimages"
+it uses, and released installers also pin the release image.  As noted above,
+release images contain `machine-os-content`, which can be a *different*
+RHCOS version.
+
+# The early pivot
+
+We do not want to require that a new bootimage is released for every update,
+and in general it can be hard to require that in every environment.  (For
+example, bare metal PXE setups)
+
+Hence, the MCO and installer combine to implement what is called "the early pivot".
+
+# Cluster bringup and upgrades
+
+Let's step through aspects of cluster bootstrap/installation and add some information
+about the early pivot.
+
+First, be sure you've read and understand [openshift/installer overview](https://github.com/openshift/installer/blob/37b99d8c9a3878bac7e8a94b6b0113fad6ffb77a/docs/user/overview.md).
+
+In this example we'll use AWS, but again this is all similar to a situation of
+booting bare metal machines via PXE or VMs in OpenStack.
+
+The openshift-installer starts, and uses the AMI it has pinned as the bootstrap node.
+
+The bootstrap node pulls the release image, which contains a reference to a newer `machine-os-content`.
+
+The installer runs the MCO in "bootstrap" mode to render Ignition configuration.
+As part of that the MCO injects a reference to that updated `machine-os-content`
+into the Ignition in the `/etc/pivot/image-pullspec` directory.
+
+The master machines boot, pulling their Ignition configs from the bootstrap
+node. As part of that `pivot.service` runs on the host (which is
+`Before=kubelet.service`, so before the cluster starts). It detects the
+reference to the updated oscontainer and pulls it, uses it to upgrade, and
```

Comment:
Will the oscontainer normally be pulled into a local registry and then made available to each node from there? Or does each node download it from a remote source?

Reply:
Today, each node downloads it - same as any other container in the payload.  Today we don't do "pull-through" to the local registry.

Reply:
Gotcha. Where's the best place for me to submit a feature request? We're working on bare-metal use cases and would prefer to not re-download these images.

Reply:
This repository is fine.